File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/axle/person/ReactivePersonEntity.java
package io.quarkus.it.mongodb.panache.axle.person;

import org.bson.codecs.pojo.annotations.BsonId;

import io.quarkus.mongodb.panache.axle.ReactivePanacheMongoEntityBase;

public class ReactivePersonEntity extends ReactivePanacheMongoEntityBase {
    @BsonId
    public Long id;
    public String firstname;
    public String lastname;
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/reactive/person/ReactivePersonEntity.kt
package io.quarkus.it.mongodb.panache.reactive.person

import io.quarkus.mongodb.panache.kotlin.reactive.ReactivePanacheMongoCompanionBase
import io.quarkus.mongodb.panache.kotlin.reactive.ReactivePanacheMongoEntityBase
import org.bson.codecs.pojo.annotations.BsonId

class ReactivePersonEntity : ReactivePanacheMongoEntityBase<ReactivePersonEntity>() {
    companion object: ReactivePanacheMongoCompanionBase<ReactivePersonEntity, Long>

    @BsonId
    var id: Long? = null
    var firstname: String? = null
    var lastname: String? = null
}